integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/Jobs/BlockletIndexInputFormat.java

```diff
@@ -123,8 +127,10 @@ public BlockletIndexInputFormat(CarbonTable table, IndexExprWrapper indexExprWra
       Cache<TableBlockIndexUniqueIdentifierWrapper, BlockletIndexWrapper> cache =
           CacheProvider.getInstance().createCache(CacheType.DRIVER_BLOCKLET_INDEX);
       private Iterator<TableBlockIndexUniqueIdentifier> iterator;
+      private Map<String, Map<String, BlockMetaInfo>> segInfoCache = new HashMap<>();
```

Comment:
please add a comment, stating why this map is required.

Reply:
Added  a comment